### PR TITLE
Add auto tool setting

### PR DESCRIPTION
**Pull request by RealIndrit**
Please XD

### CODE REVIEW

**Comment by RealIndrit** [comment]
ACTUALLY, let me fix something, realized it was retarded to have best tool be calculated if its not gonna be used anyways...

**Comment by RealIndrit** [comment]
@dominikaaaa What you think, did I miss something obvious?

**Comment by dominikaaaa** [comment]
hold on

File: src/main/java/baritone/pathing/movement/MovementHelper.java [hunk 1]
```diff
@@ -432,7 +432,9 @@ static void switchToBestToolFor(IPlayerContext ctx, IBlockState b) {
      * @param ts  previously calculated ToolSet
      */
     static void switchToBestToolFor(IPlayerContext ctx, IBlockState b, ToolSet ts, boolean preferSilkTouch) {
-        ctx.player().inventory.currentItem = ts.getBestSlot(b.getBlock(), preferSilkTouch);
+        if(Baritone.settings().autoTool.value){
```
Comment (CDAGaming) [inline src/main/java/baritone/pathing/movement/MovementHelper.java:435]: indent this.

**Comment by dominikaaaa** [comment]
closes #1334

File: src/main/java/baritone/pathing/movement/MovementHelper.java (hunk 1 above)
Comment (dominikaaaa) [reply src/main/java/baritone/pathing/movement/MovementHelper.java:435]: I'm on mobile so I missed that, oops

Comment (RealIndrit) [reply src/main/java/baritone/pathing/movement/MovementHelper.java:435]: idk it already is in my ide

**Comment by RealIndrit** [comment]
idk wtf happend but it fixed now I hope...

**Comment by dominikaaaa** [comment]
oh I didn't consider that ngl

**Comment by RealIndrit** [comment]
> Okay, but baritone's path calculation assumes autotool will be used. Also change the thing where it's like "cost of mining XXX" to take into account that the currentItem will be used and it won't autoswitch to the best one on the hotbar.

So basically I found that every mine cost depending calculation is done in the movement calculations, where its using ToolSet to find best item, all of this is + actual auto tool is depending on getBestSlot(); so i just added a line there to respect for "auto tool off" setting, which should fix the pathing calculations, due to the "current slot destruction against specific block" value being cached as forced best value instead of the actually best tool... Am I retarded or is this actually a simple and good fix?

**Comment by RealIndrit** [comment]
Also it sure does take different pathing when autotool is off, (it fails to path under ground due to not being able to use pickaxe etc)

**Comment by RealIndrit** [comment]
Ok, it works like this:
AutoTool setting: Only toggles if baritone should change to best tool when digging a block, but actually dont calucuates the cost of this movement
ignoreAutoToolMovementCost: Is the setting that actually forces Baritone to either only use current "selected slot" as best tool, or not for the movement cost calculations.

Due to how baritone  tool handling is working the first setting is completely overriden by second setting, that in other words lets people decide if they want Baritone to strictly assume their tool is the best, or if they want baritone to actually recalculate paths dependent on the tool they are using.

**Comment by scorbett123** [comment]
In your latest commit you have misspelled might as "migth" (unless you are talking about pythagorean numerology (it means  or chaldean numerology.)

**Comment by leijurv** [comment]
this is cringe but i want to fix it myself OWNED

**Pull request merged**